app/components/courses/summary_card_component.rb

```diff
@@ -208,7 +207,15 @@ def financial_incentive
     end
 
     def main_subject
-      @main_subject ||= Subject.find_by(id: course.master_subject_id)
+      @main_subject ||= course.subjects.find { |subject| subject.id == course.master_subject_id }
+    end
+
+    # rubocop:disable Lint/UselessConstantScoping
+    NullEnrichment = Struct.new(:course_length, :fee_uk_eu, :fee_international, keyword_init: true)
```

Comment:
I optin for the Null object approach. 

I will card up something to improve the tests and factories and replace this.